### ISSUE TITLE
Questions about Smart Agriculture

### ISSUE THREAD

**Issue by Allenqazwsx** [HimaxWiseEyePlus/WE_I_Plus_User_Examples#8]
Hello, I am currently using your smart agriculture example, but encountered some questions I would like to ask you, I saw in your smart agriculture sample video that you have a black himax board connected to the antenna as a receiver, but it is not specifically mentioned in the example, want to ask the board model is? Or can I take another Himx WE-I Plus antenna as a receiver? If so, will there be any changes procedurally or procedurally?

**Comment by bigcat-himax** [comment]
Hi, Allenqazwsx:
    
![Lora](https://user-images.githubusercontent.com/61962431/182532762-6a18c877-63a6-4b19-a2fe-7c2c37e3d494.png)
This is the pin description of Lora device, we poll VDD/UART_TXD/UART_RXD/GND out 

 
![connection1](https://user-images.githubusercontent.com/61962431/182532956-bc5d8062-c4ce-4bea-9c13-8acc3b3eb7bc.png)
black board here is an pin expansion for Lora and 
- VDD is the orange line
- UART_TXD is the red line
- UART_RXD is the green line
- GND is the  black line

![connection2](https://user-images.githubusercontent.com/61962431/182533234-78ea303a-10e4-44a7-816c-1f0850748b28.png)

These 4 pins out lines will connect to WE_I J3.

**Comment by Allenqazwsx** [comment]
> Hi, Allenqazwsx:
> 
> ![Lora](https://user-images.githubusercontent.com/61962431/182532762-6a18c877-63a6-4b19-a2fe-7c2c37e3d494.png) This is the pin description of Lora device, we poll VDD/UART_TXD/UART_RXD/GND out
> 
> ![connection1](https://user-images.githubusercontent.com/61962431/182532956-bc5d8062-c4ce-4bea-9c13-8acc3b3eb7bc.png) black board here is an pin expansion for Lora and
> 
> * VDD is the orange line
> * UART_TXD is the red line
> * UART_RXD is the green line
> * GND is the  black line
> 
> ![connection2](https://user-images.githubusercontent.com/61962431/182533234-78ea303a-10e4-44a7-816c-1f0850748b28.png)
> 
> These 4 pins out lines will connect to WE_I J3.

Does the antenna need to be set up in any way？Can you provide your steps？

**Comment by bigcat-himax** [comment]
Hi, Allenqazwsx:

Yes, Lora device need some AT command to setup.
As you can see in the [main.cc](https://github.com/HimaxWiseEyePlus/WE_I_Plus_User_Examples/blob/main/SmartAgriculture-example/main.cc) line 197 to line 211, these are AT commands send to Lora device for setup.

For more AT command description, please check "AT COMMAND MANUAL" document here.
https://reyax.com/products/rylr896/

**Comment by Allenqazwsx** [comment]
> Hi, Allenqazwsx:
> 
> Yes, Lora device need some AT command to setup. As you can see in the [main.cc](https://github.com/HimaxWiseEyePlus/WE_I_Plus_User_Examples/blob/main/SmartAgriculture-example/main.cc) line 197 to line 211, these are AT commands send to Lora device for setup.
> 
> For more AT command description, please check "AT COMMAND MANUAL" document here. https://reyax.com/products/rylr896/
So should i need to load another program to run your program？Or is just a matter od running exactly as per your github process？

**Comment by Allenqazwsx** [comment]
![image](https://user-images.githubusercontent.com/92588211/182616150-371e732d-6d71-4f7d-b872-da3c7b10e094.png)

**Comment by Allenqazwsx** [comment]
This is my connection，is that right？

**Comment by bigcat-himax** [comment]
Hi, Allenqazwsx:
    This is the pin description of J3
![](https://github.com/HimaxWiseEyePlus/WE_I_Plus_User_Examples/blob/main/SmartAgriculture-example/images/J3_connector.png?raw=true)
   In the figure you post, i think upper side is the sender? 
- upper left side device is connect to WE_I I2C peripheral
- upper right side device is Lora device connect to the WE_I GPIO, which we will use Software UART API described in hx_drv_tflm.h to send AT command to Lora.

For sender side, the connection is fine as shown in the post, can you please check if the sender UART console message printed correctly?

![Screenshot from 2022-08-04 08-54-34](https://user-images.githubusercontent.com/61962431/182742727-f8af191b-f0d3-48ff-ab30-850bd8ef0918.png)

red rectangle about AT command should be shown in the sender console.

**Comment by Allenqazwsx** [comment]
Does the sender need to burn your "out.img" file？

**Comment by bigcat-himax** [comment]
Yes, please follow the [flash update](https://github.com/HimaxWiseEyePlus/WE_I_Plus_User_Examples/tree/main/SmartAgriculture-example#flash-image) link to update.

**Comment by Allenqazwsx** [comment]
Sorry，in confirming one thing！Does the receiver burn the same program？

**Comment by bigcat-himax** [comment]
NO, at receiver side, you can connect any other device which able to receive Lora device's message via UART. 
For example
![receiver_side](https://user-images.githubusercontent.com/61962431/183586137-ef0255bc-517d-4d19-b18d-99d2200a5e1e.png)
we use a debug board connect between Lora UART and PC usb port.
With Tera-Term terminal, we are able to set AT command, makes Lora device to receive data.

Here are Tera-Term example scripts for you to test Lora send and receive data. 
by setting send side Lora with `AT_command_send.ttl` and you will see 'hello' message at receive side Lora when setting `AT_command_recv.ttl`.

`AT_command_recv.ttl`
timeout = 1
sendln 'AT'
wait '+OK'
sendln 'AT+FACTORY'
wait '+FACTORY'
sendln 'AT+ADDRESS=2'
wait '+OK'
sendln 'AT+CRFOP=7'
wait '+OK'

`AT_command_send.ttl`
timeout = 1
sendln 'AT'
wait '+OK'
sendln 'AT+FACTORY'
wait '+FACTORY'
sendln 'AT+CRFOP=7'
wait '+OK'
sendln 'AT+ADDRESS=1'
wait '+OK'
for i 1 999
mpause 2000
sprintf2 strtemp 'AT+SEND=2,8,HELLO%03d' i
sendln strtemp
wait '+OK'
next

**Comment by reto88** [comment]
Is it possible to adapt the code, that it only make every 10second a run, and the rest the controller and cam is in sleep mode?
